client/lib/media/utils.js

```diff
@@ -26,6 +25,17 @@ import versionCompare from 'lib/version-compare';
 
 const { uniqueId } = impureLodash;
 
+/**
+ * Taken from the official browserify 'path' polyfill: https://github.com/substack/path-browserify/blob/master/index.js
+ *
+ * Split a filename into [root, dir, basename, ext], unix version
+ * 'root' is just a slash, or nothing.
+ */
+const splitPathRe = /^(\/?|)([\s\S]*?)((?:\.{1,2}|[^\/]+?|)(\.[^.\/]*|))(?:[\/]*)$/;
+const splitPath = filename => splitPathRe.exec( filename ).slice( 1 );
+export const extname = path => splitPath( path )[ 3 ];
```

Comment:
I wouldn't expect to find these in this file. Maybe move them to a new `lib/path` or `lib/path-utils`? We already have `lib/paths` which confuses matters a bit.

Reply:
deferring this part of the vendetta for later